pupil_src/shared_modules/calibration_choreography/single_marker_plugin.py

```diff
@@ -161,9 +191,28 @@ def init_ui(self):
 
         super().init_ui()
         self.menu.append(desc_text)
-        self.menu.append(self.__ui_selector_monitor_name)
         self.menu.append(self.__ui_selector_marker_mode)
+        self.menu.append(self.__ui_selector_monitor_name)
         self.menu.append(self.__ui_slider_marker_scale)
+        self._ui_update_visibility_digital_marker_config()
+
+    def _ui_update_visibility_digital_marker_config(self):
+        try:
+            if self.__ui_slider_marker_scale is None or self.menu is None:
+                return
+        except AttributeError:
+            return
```

Comment:
When can any of this happen? Being `None` or causing AttributeError?
Especially `__ui_slider_marker_scale` seems to be never set to `None` anywhere.

Reply:
The `marker_mode` property calls `_ui_update_visibility_digital_marker_config()`. It is possible that for-what-ever-reason `init_ui()` was not called (valid if there was no ui). In this case you would not want `_ui_update_visibility_digital_marker_config()` to raise an exception

Reply:
Then I would make this more obvious by using a `self.__has_ui` which gets set to `True` in `init_ui()` and to `False` in `__init__()` and `deinit_ui()`. And then just check for this and return in `_ui_update_visibility_digital_marker_config`. Makes it obvious why wouldn't continue here.

Reply:
Or even better check for `self.__has_ui` in the `marker_mode` property before calling `_ui_update_visibility_digital_marker_config`.

Reply:
@romanroibu What do you think about this?

Reply:
Ok now I get it.
Which again makes me dislike the whole try-except because I didn't get it from reading the code  :smile: 

Yeah I don't know. Feels like the `Plugin` base class should also set `self.menu = []` or something. Maybe I've written too much C++ in the past weeks to be disturbed by Python's dynamically appearing member variables.

Reply:
:snake:

Reply:
Yeah, I don't like Python's dynamically appearing member variables, and the whole `try`/`except` construct here, but it seems cleaner than adding another `self.__has_ui` variable to keep track of the state of `self.__ui_slider_marker_scale`. Tradeoffs 🤷